src/ui/components/results/resultsheadercomponent.js

```diff
@@ -86,6 +86,15 @@ export default class ResultsHeaderComponent extends Component {
     });
   }
 
+  /**
+   * Converts an array of {@link AppliedQueryFilter}s into equivalent {@link FilterNode}s, then
+   * removes filternodes that are either blank, or have a fieldId listed in config.hiddenFields.
+   * @param {Array<AppliedQueryFilter>} nlpFilters
+   */
+  _convertNlpFiltersToFilterNodes (nlpFilters) {
+    return pruneFilterNodes(convertNlpFiltersToFilterNodes(nlpFilters), this._config.hiddenFields);
```

Comment:
Small nit, but can we have all the pruning occur in this method (both filtering out hidden fields and duplicates)? It's kinda weird how we do one step of pruning here, then another prune of nlp filters in _groupAppliedFilters.

Reply:
Pruning nlp duplicates of facets/static filters needs to happen after/within setState, since whether or not an nlp filter is a duplicate depends on the other applied filter nodes, which are updated within setState so that they can use the dynamic filters received from the response, not the facets sent in the request. I think moving the hiddenFields pruning into there makes sense though, even if it's very slightly less efficient.

Reply:
updated!